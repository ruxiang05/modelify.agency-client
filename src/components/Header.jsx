import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ReactComponent as ListIcon } from '../assets/icons/list-ul.svg';
import { ReactComponent as ModelsListIcon } from '../assets/icons/users.svg';
import { ReactComponent as AddIcon } from '../assets/icons/plus-square.svg';
import { ReactComponent as ChatIcon } from '../assets/icons/comment-alt-lines.svg';
import { ReactComponent as ProfileIcon } from '../assets/icons/user.svg';
import '../styles/header.scss';

const Header = ({ user }) => (
  <header>
    <nav>
      <ul>
        <li>
          <Link to="/jobs">
            <ListIcon />
          </Link>
        </li>
        {user.role === 'agent' && (
        <li>
          <Link to="/models">
            <ModelsListIcon />
          </Link>
        </li>
        )}
        {user.role === 'agent' && (
        <li>
          <Link
            to="/jobs/new"
          >
            <AddIcon />
          </Link>
        </li>
        )}
        <li>
          <Link to="/chats">
            <ChatIcon />
          </Link>
        </li>
        <li>
          <Link to="/profile/">
            <ProfileIcon />
          </Link>
        </li>
      </ul>
    </nav>
  </header>
);

Header.propTypes = {
  user: PropTypes.shape({}).isRequired,
};

export default Header;
