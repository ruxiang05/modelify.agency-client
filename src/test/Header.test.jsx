/* Uses methods and/or components from react and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

const user = {
  role: 'agent',
};

describe('Header', () => {
  it('should render Header', () => {
    const wrapper = shallow(<Header user={user} />);
    expect(wrapper).toMatchSnapshot();
  });
});
