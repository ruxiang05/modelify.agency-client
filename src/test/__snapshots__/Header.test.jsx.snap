// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    user={
      Object {
        "role": "agent",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <ul>
          <li>
            <Link
              replace={false}
              to="/jobs"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/models"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/jobs/new"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/chats"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/profile/"
            >
              <ForwardRef />
            </Link>
          </li>
        </ul>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <Link
              replace={false}
              to="/jobs"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/models"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/jobs/new"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/chats"
            >
              <ForwardRef />
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/profile/"
            >
              <ForwardRef />
            </Link>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/jobs"
              >
                <ForwardRef />
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/models"
              >
                <ForwardRef />
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/jobs/new"
              >
                <ForwardRef />
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/chats"
              >
                <ForwardRef />
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/profile/"
              >
                <ForwardRef />
              </Link>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/jobs"
              >
                <ForwardRef />
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ForwardRef />,
                "replace": false,
                "to": "/jobs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/models"
              >
                <ForwardRef />
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ForwardRef />,
                "replace": false,
                "to": "/models",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/jobs/new"
              >
                <ForwardRef />
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ForwardRef />,
                "replace": false,
                "to": "/jobs/new",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/chats"
              >
                <ForwardRef />
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ForwardRef />,
                "replace": false,
                "to": "/chats",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/profile/"
              >
                <ForwardRef />
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ForwardRef />,
                "replace": false,
                "to": "/profile/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
              },
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <ul>
            <li>
              <Link
                replace={false}
                to="/jobs"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/models"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/jobs/new"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/chats"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/profile/"
              >
                <ForwardRef />
              </Link>
            </li>
          </ul>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
              <Link
                replace={false}
                to="/jobs"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/models"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/jobs/new"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/chats"
              >
                <ForwardRef />
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/profile/"
              >
                <ForwardRef />
              </Link>
            </li>
          </ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/jobs"
                >
                  <ForwardRef />
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/models"
                >
                  <ForwardRef />
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/jobs/new"
                >
                  <ForwardRef />
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/chats"
                >
                  <ForwardRef />
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/profile/"
                >
                  <ForwardRef />
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/jobs"
                >
                  <ForwardRef />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ForwardRef />,
                  "replace": false,
                  "to": "/jobs",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/models"
                >
                  <ForwardRef />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ForwardRef />,
                  "replace": false,
                  "to": "/models",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/jobs/new"
                >
                  <ForwardRef />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ForwardRef />,
                  "replace": false,
                  "to": "/jobs/new",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/chats"
                >
                  <ForwardRef />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ForwardRef />,
                  "replace": false,
                  "to": "/chats",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/profile/"
                >
                  <ForwardRef />
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ForwardRef />,
                  "replace": false,
                  "to": "/profile/",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                },
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
